.github/workflows/release_fips.yaml

```diff
@@ -0,0 +1,168 @@
+name: Kubernetes
+
+on:
+  schedule:
+    # Run at 2:00 UTC every day
+    - cron: '0 2 * * *'
+  workflow_dispatch:  # Allow manual triggering
+    inputs:
+      page:
+        description: The page number to fetch
+        default: "1"
+        type: string
+      per_page:
+        description: The number of releases per page
+        default: "50"
+        type: string
+      force:
+        description: Force the build even if the release already exists
+        default: "false"
+        type: choice
+        options:
+          - "false"
+          - "true"
+      kubernetes_minor_version:
+        description: The minimum minor version of Kubernetes to build
+        type: string
+
+jobs:
+  generate-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.get-releases.outputs.matrix }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Get stable K8s releases
+        id: get-releases
+        run: |
+          set -euo pipefail
+          MIN_VERSION=${{ inputs.kubernetes_minor_version }}
+          MIN_VERSION=${MIN_VERSION:-33}
+
+          # Get stable releases >= 1.MIN_VERSION where version file exists
+          MATRIX=$(curl -s "https://api.github.com/repos/kubernetes/kubernetes/releases?page=${{ inputs.page || 1 }}&per_page=${{ inputs.per_page || 50 }}" | \
+            jq -r '.[] | select(.tag_name | test("^v[0-9]+\\.[0-9]+\\.[0-9]+$")) | .tag_name' | \
+            sort -V | \
+            awk -F'[.v]' -v min=$MIN_VERSION '$2==1 && $3>=min' | \
+            while read version; do
+              trimmed=$(echo $version | sed -E 's/\.[0-9]+$//')
+              [ -f "./hack/kubernetes-${trimmed}" ] && echo "$version"
+            done | \
+            jq -R -s -c 'split("\n") | map(select(length > 0)) | {version: .}')
```

Comment:
I would rather filter out the already existing images here instead of within the matrix since this is running every day so we would basically start a bazillion jobs even if they are short running